src/components/core/providers/use-ctw.ts

```diff
@@ -28,6 +28,11 @@ export function useCTW() {
       builderId: context.builderId ?? claimsBuilderId(authToken) ?? "",
       contextBuilderId: context.builderId,
       fetchFromFqs: getFetchFromFqs(context.env, authToken, context.builderId),
+      onResourceSave: (resource, action) => {
+        if (context.onResourceSave) {
+          context.onResourceSave(resource, action);
+        }
+      },
```

Comment:
nit: This is a little gross, alternatively you could make `onResourceSave` optional, and this would simply be:
```suggestion
      onResourceSave: context.onResourceSave,
```

Then callers would do `context.onResourceSave?.(resource, action)` to call it.

FYI, I don't have a strong opinion here.

Reply:
Yeah I want it to be optional for the consumers of the library but internally callers don't need to know whether the consumer actually implemented it or not, it's something the context provides always.